### PR TITLE
Implementation of subcloud add without upload

### PR DESCRIPTION
**Pull request by sharbhar**
Added a fact for helm-chart and overrides to get DM artifacts from subcloud 
When DM artifacts are not uploaded to /opt/platform/deploy/

Test Plan:
PASS: Test the DM artifacts are picked up from subcloud in DM install. 
PASS: Deployment is successful.
PASS: DM pod is running in the subcloud.

### CODE REVIEW

**Comment by yjian118** [comment]
Need more info about the tests passed in detail:
1 Is this change tested in stand-alone system?
2 Is this change tested in DC system?
3 Is this change tested with "dcmanager subcloud reconfig"?

**Comment by sharbhar** [comment]
> Need more info about the tests passed in detail: 1 Is this change tested in stand-alone system? 2 Is this change tested in DC system? 3 Is this change tested with "dcmanager subcloud reconfig"?

 1 Is this change tested in stand-alone system? 
    >> No, not tested with stand-alone system.
 2 Is this change tested in DC system? 
   >> Yes, tested with DC-system using dc-libvrt set-up.
 3 Is this change tested with "dcmanager subcloud reconfig"?
   >> No, not tested with subcloud reconfig.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml [hunk 1]
```diff
@@ -89,6 +89,11 @@
           helm_chart_overrides: "/home/{{ ansible_ssh_user }}/{{ helm_chart_overrides | basename }}"
         when: helm_chart_overrides is defined
 
+      - set_fact:
+          helm_chart_overrides: "/usr/local/share/applications/overrides/wind-river-cloud-platform-deployment-manager-overrides-subcloud.yaml"
```
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:93]: Can we check the existence of these two files right after setting the facts? And fail the playbook with a meaningful error message?

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml [hunk 2]
```diff
@@ -89,6 +89,11 @@
...
+          manager_chart: "/usr/local/share/applications/helm/wind-river-cloud-platform-deployment-manager-*.tgz"
+        when: ("subcloud" in get_distributed_cloud_role.stdout and "not uploaded" in user_uploaded_artifacts)
```
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:95]: Note: the user_uploaded_artifacts may not be defined, and the playbook will throw out an error in this case. Need to test this playbook on standalone system

Comment (sharbhar) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:95]: run the playbook without user_uploaded_artifacts

image

Comment (sharbhar) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:95]: executed the playbook with user_uploaded_artifacts defined to "not uploaded"

image


image

Comment (yjian118) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:95]: Thanks for the re-test, we can hold this PR until the change in the dc repo finalized

Comment (yjian118) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:95]: And the "dcmanager subcloud reconfig" test is required once the change in DC repo finalized

Comment (sselvara1) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:95]: As discussed in scrum, playbook changes can be merged independent of the dc repo.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml (hunk 1 above)
Comment (sharbhar) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:93]: Tested the existence 
image

Failing the playbook by deleting the overrides file.
image

**Review by sselvara1 — COMMENTED** [review]
LGTM

**Review by hugonicodemos — APPROVED** [review]
LGTM

**Review by ttakenak — APPROVED** [review]

**Pull request merged**